### ISSUE TITLE
Maven Plugin: UnsupportedClassVersionError when maven compiles with Java 17 while JAVA_HOME points to Java 8

### ISSUE THREAD

**Issue by gemiusz** [gatling/gatling#4285]
gatling-charts-highcharts: 3.8.2
gatling-maven-plugin: 4.2.1
OS: Windows 10 pro
JDK: 17.0.3.1

<maven.compiler.release>11</maven.compiler.release>   or  <maven.compiler.release>17</maven.compiler.release>

Bug generated on [Gatling plugin for Maven - Java demo project](https://github.com/gatling/gatling-maven-plugin-demo-java)

After `mvn gatling:test` gettling below StackTrace:
```
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------< io.gatling.demo:gatling-maven-plugin-demo-java >-----------
[INFO] Building gatling-maven-plugin-demo-java 3.8.2
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] >>> gatling-maven-plugin:4.2.1:test (default-cli) > test-compile @ gatling-maven-plugin-demo-java >>>
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ gatling-maven-plugin-demo-java ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory [MASKED_DIR]\gatling-maven-plugin-demo-java\src\main\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ gatling-maven-plugin-demo-java ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ gatling-maven-plugin-demo-java ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 4 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ gatling-maven-plugin-demo-java ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 4 source files to [MASKED_DIR]\gatling-maven-plugin-demo-java\target\test-classes
[INFO] 
[INFO] <<< gatling-maven-plugin:4.2.1:test (default-cli) < test-compile @ gatling-maven-plugin-demo-java <<<
[INFO] 
[INFO] 
[INFO] --- gatling-maven-plugin:4.2.1:test (default-cli) @ gatling-maven-plugin-demo-java ---
21:53:00.329 [ERROR] i.g.a.Gatling$ - Run crashed
java.lang.UnsupportedClassVersionError: computerdatabase/ComputerDatabaseSimulation has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:473)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:405)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at io.gatling.app.classloader.SimulationClassLoader.$anonfun$simulationClasses$2(SimulationClassLoader.scala:49)
	at scala.collection.immutable.List.map(List.scala:246)
	at scala.collection.immutable.List.map(List.scala:79)
	at io.gatling.app.classloader.SimulationClassLoader.simulationClasses(SimulationClassLoader.scala:49)
	at io.gatling.app.Selection$Selector.$anonfun$selection$1(Selection.scala:49)
	at scala.Option.getOrElse(Option.scala:201)
	at io.gatling.app.Selection$Selector.selection(Selection.scala:44)
	at io.gatling.app.Selection$.apply(Selection.scala:36)
	at io.gatling.app.Runner.run(Runner.scala:55)
	at io.gatling.app.Gatling$.start(Gatling.scala:90)
	at io.gatling.app.Gatling$.fromArgs(Gatling.scala:52)
	at io.gatling.app.Gatling$.main(Gatling.scala:40)
	at io.gatling.app.Gatling.main(Gatling.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.gatling.plugin.util.MainWithArgsInFile.runMain(MainWithArgsInFile.java:53)
	at io.gatling.plugin.util.MainWithArgsInFile.main(MainWithArgsInFile.java:34)
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.gatling.plugin.util.MainWithArgsInFile.runMain(MainWithArgsInFile.java:53)
	at io.gatling.plugin.util.MainWithArgsInFile.main(MainWithArgsInFile.java:34)
Caused by: java.lang.UnsupportedClassVersionError: computerdatabase/ComputerDatabaseSimulation has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:473)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:405)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at io.gatling.app.classloader.SimulationClassLoader.$anonfun$simulationClasses$2(SimulationClassLoader.scala:49)
	at scala.collection.immutable.List.map(List.scala:246)
	at scala.collection.immutable.List.map(List.scala:79)
	at io.gatling.app.classloader.SimulationClassLoader.simulationClasses(SimulationClassLoader.scala:49)
	at io.gatling.app.Selection$Selector.$anonfun$selection$1(Selection.scala:49)
	at scala.Option.getOrElse(Option.scala:201)
	at io.gatling.app.Selection$Selector.selection(Selection.scala:44)
	at io.gatling.app.Selection$.apply(Selection.scala:36)
	at io.gatling.app.Runner.run(Runner.scala:55)
	at io.gatling.app.Gatling$.start(Gatling.scala:90)
	at io.gatling.app.Gatling$.fromArgs(Gatling.scala:52)
	at io.gatling.app.Gatling$.main(Gatling.scala:40)
	at io.gatling.app.Gatling.main(Gatling.scala)
	... 6 more
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.796 s
[INFO] Finished at: 2022-07-14T21:53:00+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.gatling:gatling-maven-plugin:4.2.1:test (default-cli) on project gatling-maven-plugin-demo-java: Gatling failed.: ForkException -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

Process finished with exit code 1

```

**Comment by slandelle** [comment]
This error means Gatling is still running with Java 8.
Could you please share your `PATH` and your `JAVA_HOME`?

**Comment by gemiusz** [comment]
You are right - my mistake. 
I forgot about `JAVA_HOME` but I wondering why when IntelliJ starting Maven using Java from project (JDK17),  gatling-maven-plugin uses this from `JAVA_HOME` (JDK1.8) in the same build.
After change value `JAVA_HOME` to  `JDK17` everything is working. 
Sorry for the confusion.
Issue can be closed.

**Comment by slandelle** [comment]
No, I think it’s a real issue and our maven plugin should give precedence to the Java installation that was used to launch maven. Our behavior is not aligned with what maven-compiler-plugin does.
Going to fix.
Thanks for reporting !

**Comment by slandelle** [comment]
* maven: fixed in gatling-maven-plugin 4.2.1 (released, will be sync on maven central in ~20 minutes)
* bundle: will be fixed in 3.8.3
